### PR TITLE
ESLint fixes

### PR DESCRIPTION
**Pull request by mikemand**
_And other various code smells..._

I should have done this PR a long while ago, but I haven't had very much time for the last almost year.

There are a few various bug fixes (like the MediaList icons not updating correctly when you change to a folder, the MediaTransformer not sending all the necessary data, etc.), but most everything is just me opening each Vue file with ESLint active in my IDE and changing anything it says to change.

I left all my commits as they were authored (no rebasing or amending), but I can squash them if you want.

### CODE REVIEW

**Comment by mikemand** [comment]
This is a precursor to me starting work on upgrading to Laravel 6.

**Comment by mikemand** [comment]
Oh! I forgot to mention I upgraded to Laravel Mix 4. That's probably the biggest thing to take note of. You'll need to install NPM packages again if you want to compile assets.

File: composer.json [hunk 1]
```diff
@@ -17,18 +17,18 @@
...
-        "idavoll/setting-module": "4.0.x-dev",
-        "idavoll/media-module": "4.0.x-dev",
-        "idavoll/page-module": "4.0.x-dev",
-        "idavoll/menu-module": "4.0.x-dev",
-        "idavoll/workshop-module": "4.0.x-dev",
-        "idavoll/translation-module": "4.0.x-dev",
-        "idavoll/tag-module": "4.0.x-dev",
-        "idavoll/flatly-theme": "4.0.x-dev",
-        "idavoll/adminlte-theme": "4.0.x-dev"
+        "idavoll/adminlte-theme": "^4.0.x-dev",
```
Comment (nWidart) [inline composer.json]: This seem to fail in travis.

**Comment by nWidart** [comment]
This looks very good! Thank you so much 🙏 
One thing is the composer issue where travis isn't able to pull those versions, not sure if that's because of the added caret, or because it needs to run one subsplit after the merge.

**Comment by mikemand** [comment]
@nWidart I'm not sure what's going on with that, but if I use just `^4.0` in my composer.json it works. Should I make the change here too?

**Comment by mikemand** [comment]
Ah, ok. It was the caret. Fixed. 😃

**Comment by nWidart** [comment]
Sweet thanks! I think it's because it's a branch alias, not a usual composer version.

**Comment by mikemand** [comment]
It's weird that I can use `^4.0`. Maybe that's part of Composer parsing the branch alias? Not sure.

**Pull request merged**